fixed spacing in navbar and footer

hars-21 opened the pull request:
I have fixed the spacing and alignment issue in the nav and the footer components. 
fixes: #346 

## 1. Improved spacing in nav: 

previously - <img width="1280" height="120" alt="Image" src="https://github.com/user-attachments/assets/d097f836-9769-4a10-b431-bce0b8aeec8c" />

after improvement -
<img width="1263" height="140" alt="image" src="https://github.com/user-attachments/assets/cb88dbfb-1bbc-4763-919e-1135e6d6a378" />

## 2. Improved alignment in footer:

previously - <img width="415" height="615" alt="Image" src="https://github.com/user-attachments/assets/4d6f32d8-2208-4dad-a146-ed43ca6abb16" />

after improvement - 
<img width="395" height="653" alt="image" src="https://github.com/user-attachments/assets/947f0470-1931-4fa5-be19-d80781a7030d" />

jsilvela reviewed (changes requested):
Hello:

1. please sign your commit. Have a look at the failing DCO check, which includes instructions on fixing your commit
2. I took your PR for a spin. The footer seems improved. The nav bar does not. See grab:

<img width="1763" height="209" alt="Screenshot 2025-09-01 at 10-03-52 CloudNativePG - PostgreSQL Operator for Kubernetes" src="https://github.com/user-attachments/assets/f6761d33-4f6d-460d-b09f-6404d21a491c" />

hars-21 commented:
I have signed my commits and DCO check is now approved. Also, I have fixed navbar at my end.
Result: 
<img width="1280" height="123" alt="image" src="https://github.com/user-attachments/assets/13ced68e-d62a-4ae4-aba4-88f653fdbb53" />


Please let me know if the problem still persists. Thanks!

jsilvela reviewed (changes requested):
Not fixed. Not at all.
The problem with the navbar persists.
Looking at the nav partial, I see the same change as you had made originally, i.e. using gap-10 instead of gap-14.
So, you changed nothing but merged from main and messed up your commit history.

hars-21 committed:
fixed spacing in nav and footer

Signed-off-by: Harshil <harshilgupta.2005@gmail.com>

jsilvela reviewed (changes requested):
Are you testing your PR with different sizes of window?
Given the problem is the collision of the first item with the logo and the last item with the github icons, I don't understand your insistence on changing the inter-item gap.

Two different window sizes that produce collisions with your fix.

<img width="1516" height="181" alt="Screenshot 2025-09-04 at 17-29-04 CloudNativePG - PostgreSQL Operator for Kubernetes" src="https://github.com/user-attachments/assets/58e92581-d1d9-4998-ae89-062286a1c9d0" />
<img width="1707" height="176" alt="Screenshot 2025-09-04 at 17-27-54 CloudNativePG - PostgreSQL Operator for Kubernetes" src="https://github.com/user-attachments/assets/312c5733-f29f-463e-bcb9-66adf7b67a49" />

jsilvela commented:
I'm not convinced. Your fix is ballooning in complexity, and you're solving the collision by removing the menu altogether.
I don't understand why the nav cannot be fixed surgically in one line.

hars-21 commented:
Sir this is the second approach in which I have added javascript to resize the github star button based on screen size. It was necessary as it could not be solved through simple CSS. 
I have removed the hamburger menu from tablet and reduced the size of logo and github button. 
please tell me if this would work.

hars-21 commented:
> For my understanding, and with no code necessary, just use your own words: how would you fix the collision in only one line?

We can apply gap to the main container which will introduce spacing between logo, links and github badge. Or provide padding or margin to left and right for removing the collision.

Are these approaches good or is there any other concept which I am unable to think of.

jsilvela commented:
OK, then, margin or padding will avoid the collision. Why haven't you done this before?
Much simpler than tacking on Javascript or doing window-dependent resizing/gapping wouldn't you say?

And, I don't think it's very coherent to change the font size for the navbar progressively while the rest of the text on the page remains the same size.
I do agree with reducing the inter-item gap slightly. 14px was a bit much.
Keep it simple and short, and fix the issues pointed out in the issue only.

jsilvela commented:
> while it solves the collision issue but it also creates other ui issues such as distorted logo, improper spacing and overflowing.

The padding does not introduce those issues. Those issues exist already in production.
Since the problems the Issue complained of were the footer spacing and the collision of the navbar, I think you should do the easiest fix. And as I mentioned, I think reducing the inter-item gap from 14 to, say, 10 px, is also relevant as part of this.

Other issues can be raised in new tickets, and fixed in other PRs.
Given the website is likely to go through some redesign soon, please keep the PR small and to-the-point.

hars-21 committed:
navbar collision fixed

Signed-off-by: Harshil <harshilgupta.2005@gmail.com>
